scripts/installer/windows/scripts/generate_dependency.py

```diff
@@ -34,28 +44,40 @@ def extract_package_names(packages):
 def get_package_dependencies(package_name_list):
     dependencies = []
     for package_name in package_name_list:
-        result = subprocess.run(['pip', 'show', package_name], stdout=subprocess.PIPE)
+        if (is_tool('conda')):
+            result = subprocess.run('conda activate root | pip show {}'.format(package_name), shell=True, stdout=subprocess.PIPE)
+        else:
+            result = subprocess.run(['pip', 'show', package_name], stdout=subprocess.PIPE)
+        print("---" + package_name)
+        print(result.stdout)
         lines = result.stdout.decode('utf-8', errors="ignore").splitlines()
         for line in lines:
             if line.startswith('Requires'):
                 dependency = line.split(': ')[1].split(', ')
                 if dependency != ['']:
                     dependencies.extend(dependency)
                 break
+
+    dependencies = [dependency for dependency in dependencies if not dependency.startswith('promptflow')]
     return dependencies
 
 
 if __name__ == '__main__':
     dependencies = []
-    install_requires, extras_requires = extract_requirements('../../../../src/promptflow/setup.py')
+    install_requires, extras_requires = extract_requirements(get_git_base_dir() / 'src/promptflow/setup.py')
     install_requires_names = extract_package_names(install_requires)
     dependencies.extend(install_requires_names)
 
     for key in extras_requires:
         extras_require_names = extract_package_names(extras_requires[key])
         dependencies.extend(extras_require_names)
+    dependencies = list(set(dependencies))
```

Comment:
It seems we also need to get package in each toml file or maintain a msi seperate toml if pyinstaller support.
I will look into the it later.